### ISSUE TITLE
Upgrading from 2.0.0 to 2.0.1 fails on DBs with continuous aggregates

### ISSUE THREAD

**Issue by LeeHampton** [timescale/timescaledb#2901]
**Relevant system information:**
 - OS: Linux (official timescaledb docker image) 
 - PostgreSQL version (output of `postgres --version`): 12.5
 - TimescaleDB version (output of `\dx` in `psql`): 2.0.0
 - Installation method: using Docker

**Describe the bug**
When we run `ALTER EXTENSION UPDATE timescaledb` using the latest 2.0.1 Docker image on a DB that uses continuous aggregates, we get the following output and are unable to successfully upgrade.:

```
ERROR:  functionality not supported under the current "timescale" license
HINT:  Upgrade your license to 'timescale' to use this free community feature.
CONTEXT:  SQL statement "ALTER MATERIALIZED... Show more
```

cc @feikesteenbergen or @alexeyklyukin for additional context.

**Comment by hgiasac** [comment]
I also face this issue. We have to update license manually to enable Timescaledb features

```sql
ALTER SYSTEM SET timescaledb.license = 'timescale';
```

**Comment by samuelstroschein** [comment]
Is there another method that does not require SUPERUSER permissions? Managed Databases from Digital Ocean and Google Cloud etc. do not grant SUPERUSER permissions under any circumstances.

**Comment by jankatins** [comment]
We've a similar case. We upgrade timescaledb by first adding `timescaledb.disable_load` into `postgres.conf` and restarting. Afterwards, we try to update timescale with a non-superuser (extension upgrade happens via pgextwlist, so via mostly superuser rights -> see line 10-7 how we set the user!). This results in an error (from pg log):

```
[10-1]  ERROR:  functionality not supported under the current "timescale" license
[10-2]  HINT:  Upgrade your license to 'timescale' to use this free community feature.
[10-3]  CONTEXT:  SQL statement "ALTER MATERIALIZED VIEW motion_aggregate SET (timescaledb.materialized_only='f') "
[10-4] 	PL/pgSQL function inline_code_block line 25 at EXECUTE
[10-5]  STATEMENT:
[10-6] 	            SELECT pg_catalog.set_config('search_path', 'public', false);
[10-7] 	            SET ROLE "tsdbadmin";
[10-8] 	            ALTER EXTENSION timescaledb UPDATE
```

But running the `ALTER EXTENSION` it as superuser also fails (this is run without opening another connection!):

```
[11-1]  ERROR:  could not open relation with OID 1675407
[11-2]  CONTEXT:  SQL statement "SELECT extversion                 FROM pg_extension WHERE extname = 'timescaledb'"
[11-4]  STATEMENT:  ALTER EXTENSION timescaledb UPDATE
[11-3] 	PL/pgSQL function inline_code_block line 8 at SQL statement
```
(there was load on the DB, which actually triggered a `ERROR:  function "continuous_agg_invalidation_trigger" is not supported under the current "timescale" license` while `INSERT`ing into a table, not sure if that makes a difference)

In a second case, I have this (unfortunately, only the client log):

Without superuser, but the "elevation" provided by pgextwlist
```
ERROR: functionality not supported under the current "timescale" license
HINT:  Upgrade your license to \'timescale\' to use this free community feature.
CONTEXT:  SQL statement "ALTER MATERIALIZED VIEW pages_aggregation SET (timescaledb.materialized_only=false)
PL/pgSQL function inline_code_block line 24 at EXECUTE
```

With superuser (again, same connection!):
```
ERROR: could not open relation with OID 152118
CONTEXT:  SQL statement "SELECT extversion                 FROM pg_extension WHERE extname = 'timescaledb'"
```

**Comment by mfundul** [comment]
@LeeHampton do you by any chance use `timescaledb.disable_load` as well?

**Comment by mfundul** [comment]
Closing this since @LeeHampton verified this has been resolved after upgrading to version 2.0.1
@jankatins if your issue still persists with version 2.4.1 you should create a new issue with your description above:
https://github.com/timescale/timescaledb/issues/2901#issuecomment-891123076
